### ISSUE TITLE
`IndexDefect` when encountering `mixin` in generic code

### ISSUE THREAD

**Issue by arnetheduck** [nim-lang/Nim#22012]
### Description

```
compiler/nim.nim(152) nim
compiler/nim.nim(108) handleCmdLine
compiler/main.nim(278) mainCommand
compiler/main.nim(248) compileToBackend
compiler/main.nim(104) commandCompileToC
compiler/modules.nim(133) compileProject
compiler/modules.nim(53) compileModule
compiler/passes.nim(186) processModule
compiler/passes.nim(74) processTopLevelStmt
compiler/sem.nim(656) myProcess
compiler/sem.nim(624) semStmtAndGenerateGenerics
compiler/semstmts.nim(2443) semStmt
compiler/semexprs.nim(1108) semExprNoType
compiler/semexprs.nim(3194) semExpr
compiler/importer.nim(341) evalImport
compiler/importer.nim(311) impMod
compiler/importer.nim(284) myImportModule
compiler/modules.nim(82) importModule
compiler/modules.nim(53) compileModule
compiler/passes.nim(186) processModule
compiler/passes.nim(74) processTopLevelStmt
compiler/sem.nim(656) myProcess
compiler/sem.nim(624) semStmtAndGenerateGenerics
compiler/semstmts.nim(2443) semStmt
compiler/semexprs.nim(1108) semExprNoType
compiler/semexprs.nim(3194) semExpr
compiler/importer.nim(343) evalImport
compiler/importer.nim(311) impMod
compiler/importer.nim(284) myImportModule
compiler/modules.nim(82) importModule
compiler/modules.nim(53) compileModule
compiler/passes.nim(186) processModule
compiler/passes.nim(74) processTopLevelStmt
compiler/sem.nim(656) myProcess
compiler/sem.nim(624) semStmtAndGenerateGenerics
compiler/semstmts.nim(2443) semStmt
compiler/semexprs.nim(1108) semExprNoType
compiler/semexprs.nim(3194) semExpr
compiler/importer.nim(343) evalImport
compiler/importer.nim(311) impMod
compiler/importer.nim(284) myImportModule
compiler/modules.nim(82) importModule
compiler/modules.nim(53) compileModule
compiler/passes.nim(186) processModule
compiler/passes.nim(74) processTopLevelStmt
compiler/sem.nim(656) myProcess
compiler/sem.nim(624) semStmtAndGenerateGenerics
compiler/semstmts.nim(2443) semStmt
compiler/semexprs.nim(1108) semExprNoType
compiler/semexprs.nim(3179) semExpr
compiler/semstmts.nim(2205) semProc
compiler/semstmts.nim(1911) semProcAux
compiler/semstmts.nim(1642) semProcAnnotation
compiler/sem.nim(502) semMacroExpr
compiler/sem.nim(441) semAfterMacroCall
compiler/semexprs.nim(3179) semExpr
compiler/semstmts.nim(2205) semProc
compiler/semstmts.nim(2138) semProcAux
compiler/semgnrc.nim(511) semGenericStmt
compiler/semgnrc.nim(501) semGenericStmt
compiler/semgnrc.nim(494) semGenericStmt
compiler/semgnrc.nim(50) semGenericStmtScope
compiler/semgnrc.nim(501) semGenericStmt
compiler/semgnrc.nim(345) semGenericStmt
compiler/semgnrc.nim(501) semGenericStmt
compiler/semgnrc.nim(382) semGenericStmt
compiler/semgnrc.nim(501) semGenericStmt
compiler/semgnrc.nim(224) semGenericStmt
compiler/semtempl.nim(115) semMixinStmt
compiler/lookups.nim(56) considerQuotedIdent
Error: unhandled exception: index out of bounds, the container is empty [IndexDefect]
```

In a slightly complex case of generic code using a mixin, hitting the above error with nim 1.6.12 - `nkOpenSymChoice` without childen, as found here:
https://github.com/nim-lang/Nim/blob/1aa9273640c0c51486cf3a7b67282fe58f360e91/compiler/lookups.nim#L56

in particular, the mixin symbol doesn't appear anywhere else in the codebase except the mixin statement.

### Nim Version

1.6.12

**Comment by metagn** [comment]
Is the mixin maybe on a non-overloadable symbol, like a type?

**Comment by arnetheduck** [comment]
> Is the mixin maybe on a non-overloadable symbol, like a type?

it doesn't exist at all in the source code apart from being an argument to `mixin`

**Comment by arnetheduck** [comment]
it can reliably be reproduced by checking out https://github.com/status-im/nimbus-eth2/pull/5032/ and re-adding `beaconBlocksByRange` that the PR removes (ie running `make` to build)

**Comment by metagn** [comment]
The fix should be pretty easy but I cannot figure out a test case or how to encounter it at all for the life of me

**Comment by arnetheduck** [comment]
> The fix should be pretty easy but I cannot figure out a test case or how to encounter it at all for the life of me

haha, yeah, normally I strive to minimise test cases, but this time .. that code is quite the maze with all the macros and generics

**Comment by metagn** [comment]
Never mind, I was only trying `mixin` with just 1 undeclared symbol which was fixed (#18968), the issue appears when other, existing symbols are in the same mixin statement

```nim
import macros

expandMacros:
  proc foo[T](): int =
    mixin nothing, add
    123

  echo foo[int]()
```

```
proc foo[T](): int =
  mixin <<0th child missing for nkOpenSymChoice >>, add
  123

echo [foo()]
fatal.nim(54)            sysFatal
Error: unhandled exception: index out of bounds, the container is empty [IndexDefect]
```
